core/src/main/java/com/homeaway/streamingplatform/db/dao/impl/StreamDaoImpl.java

```diff
@@ -153,12 +165,16 @@ public void upsertStream(Stream stream) {
                 avroStream.setCreated(System.currentTimeMillis());
                 key = AvroStreamKey.newBuilder().setStreamName(avroStream.getName()).build();
             }
+
             verifyAndUpsertTopics(stream);
             kafkaProducer.log(key, avroStream);
             log.info("Stream upserted for {}", stream.getName());
         } catch (IllegalArgumentException e) {
             log.error("caught an illegal argument exception");
             throw new IllegalArgumentException(e);
+        } catch (SchemaManagerException e) {
+            log.error("caught an exception during schema registration", e);
```

Comment:
why catch and rethrow just to log?
Don't we log at the top of the callchain? If not, we should.

Reply:
yes we do, I removed the superfluous catch block